Benchmarks: re-enable profiling-te configuration.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #44218 [TensorExpr] Re-enable tests.
* **#44212 Benchmarks: re-enable profiling-te configuration.**

Differential Revision: [D23544563](https://our.internmc.facebook.com/intern/diff/D23544563)

Commit:
Benchmarks: re-enable profiling-te configuration.

[ghstack-poisoned]

wconstab reviewed (approved):
i think just merge for now, we could at some point discuss what the strategy should be for benchmarking additional (non-master) configs, at some point as we add workloads and runtime it could become an issue, but not yet.  any other thoughts?

bertmaher commented on benchmarks/fastrnns/test_bench.py:
```diff
@@ -13,7 +13,7 @@ def pytest_generate_tests(metafunc):
     if metafunc.cls.__name__ == "TestBenchNetwork":
         metafunc.parametrize('net_name', all_nets, scope="class")
-        metafunc.parametrize("executor_and_fuser", ["legacy-old"], scope="class")
+        metafunc.parametrize("executor_and_fuser", ["legacy-old", "profiling-te"], scope="class")
```
Even though I added this in the first place, I feel uncomfortable with it because I was seeing some odd behavior when running a graph through both the legacy and profiling executor in the same script.  As far as I could tell, switching the executor didn't actually take effect (which is bizarre, and inexplicable, so *maybe* I was doing something wrong).

All this to say, how would you feel about taking the `conftest.py` approach from pytorch/benchmark and using a custom command line option for the fuser/executor, rather than adding it as a fixture?  Alternatively, convince me that I was temporarily insane when I thought I was seeing the wrong executor being used!

Commit:
Update on "Benchmarks: re-enable profiling-te configuration."


Differential Revision: [D23544563](https://our.internmc.facebook.com/intern/diff/D23544563)

[ghstack-poisoned]

wconstab reviewed (approved):
lgtm although are you going to rename legacy to old in the case of executor?

ZolotukhinM commented:
> lgtm although are you going to rename legacy to old in the case of executor?

Probably not: so far we've been consistently using "legacy" for the executor and "old" for the fuser. It might be a questionable naming, but that's what it is now :)

ngimel commented:
This broke a lot of tests with 
```
Sep 07 01:20:40 Legend:
Sep 07 01:20:40   Outliers: 1 Standard Deviation from Mean; 1.5 IQR (InterQuartile Range) from 1st Quartile and 3rd Quartile.
Sep 07 01:20:40   OPS: Operations Per Second, computed as 1 / Mean
Sep 07 01:20:40 ======================== 24 passed, 1 warning in 36.21s ========================
Sep 07 01:20:41 + python benchmarks/upload_scribe.py --pytest_bench_json benchmarks/.data/fastrnns_legacy_old.json
Sep 07 01:20:41 Traceback (most recent call last):
Sep 07 01:20:41   File "benchmarks/upload_scribe.py", line 137, in <module>
Sep 07 01:20:41     benchmark_uploader.post_pytest_benchmarks(json_data)
Sep 07 01:20:41   File "benchmarks/upload_scribe.py", line 99, in post_pytest_benchmarks
Sep 07 01:20:41     executor, fuser = b['params']['executor_and_fuser'].split('-')
Sep 07 01:20:41 KeyError: 'executor_and_fuser'
Sep 07 01:20:41 + cleanup
Sep 07 01:20:41 + retcode=1
Sep 07 01:20:41 + set +x
Sep 07 01:20:41 =================== sccache compilation log ===================
Sep 07 01:20:41 =========== If your build fails, please t
```
,reverting
